src/Kentico.Xperience.AiraUnified/Chat/Models/AiraUnifiedChatMessageViewModel.cs

```diff
@@ -53,4 +53,29 @@ internal sealed class AiraUnifiedChatMessageViewModel
     /// Application insights for current message.
     /// </summary>
     public InsightsResponseModel? Insights { get; set; }
+
+    /// <summary>
+    /// Indicates if this is a system message containing insights data.
+    /// </summary>
+    public bool IsInsightsMessage => Role == "system" && !string.IsNullOrEmpty(Message);
```

Comment:
This implies all system messages are insights messages, is this really the case?